templates/explore/code.tmpl

```diff
@@ -16,13 +16,22 @@
                 <h3>
                     {{.i18n.Tr "explore.code_search_results" (.Keyword|Escape) | Str2html }}
                 </h3>
+				<div>
+					{{range $term := .SearchResultLanguages}}
+					<a class="ui basic label" href="{{$.AppSubURL}}/explore/code?l={{$term.Language}}&q={{$.Keyword}}">
+						<i class="color-icon" style="background-color: {{$term.Color}}"></i>
+						{{$term.Language}}
+						<div class="detail">{{$term.Count}}</div>
+					</a>
+					{{end}}
+				</div>
                 <div class="repository search">
                     {{range $result := .SearchResults}}
                         {{$repo := (index $.RepoMaps .RepoID)}}
                         <div class="diff-file-box diff-box file-content non-diff-file-content repo-search-result">
                             <h4 class="ui top attached normal header">
                                 <span class="file"><a rel="nofollow" href="{{EscapePound $repo.HTMLURL}}">{{$repo.FullName}}</a> - {{.Filename}}</span>
-                                <a class="ui basic grey tiny button" rel="nofollow" href="{{EscapePound $repo.HTMLURL}}/src/branch/{{$repo.DefaultBranch}}/{{EscapePound .Filename}}">{{$.i18n.Tr "repo.diff.view_file"}}</a>
+                                <a class="ui basic grey tiny button" rel="nofollow" href="{{EscapePound $repo.HTMLURL}}/src/commit/{{$result.CommitID}}/{{EscapePound .Filename}}">{{$.i18n.Tr "repo.diff.view_file"}}</a>
```

Comment:
`CommitID` maybe empty?

Reply:
How it could be empty? It's always indexed with commit id